src/TLuaInterpreter.cpp

```diff
@@ -9207,57 +9207,152 @@ int TLuaInterpreter::setRoomChar(lua_State* L)
+                       luaL_typename(L, 2));
+        return lua_error(L);
     } else {
         c = lua_tostring(L, 2);
     }
+
     Host& host = getHostFromLua(L);
     TRoom* pR = host.mpMap->mpRoomDB->getRoom(id);
     if (!pR) {
-        lua_pushstring(L, "setRoomChar: room ID does not exist");
-        lua_error(L);
-        return 1;
+        lua_pushnil(L);
+        lua_pushfstring(L, "room with id %d does not exist", id);
+        return 2;
     } else {
         if (c.size() >= 1) {
-            pR->c = c[0];
+            if (c[0] == ' ') {
+                pR->mSymbol.clear();
+                lua_pushboolean(L, true);
+                return 1;
+            } else if (c[0] >= 33) {
+                pR->mSymbol = c[0];
+                lua_pushboolean(L, true);
+                return 1;
+            } else {
+                lua_pushnil(L);
+                lua_pushfstring(L, "invalid first character code: %d in supplied room character", c[0]);
+                return 2;
+            }
+        } else {
+            // Allow an empty string to be used to clear the symbol:
+            pR->mSymbol.clear();
+            lua_pushboolean(L, true);
+            return 1;
         }
     }
-    return 0;
 }
 
 int TLuaInterpreter::getRoomChar(lua_State* L)
 {
     int id;
     if (!lua_isnumber(L, 1)) {
-        lua_pushstring(L, "getRoomChar: wrong argument type");
-        lua_error(L);
-        return 1;
+        lua_pushfstring(L, "getRoomChar: bad argument #1 type (room id as number expected, got %s!)",
+                       luaL_typename(L, 1));
+        return lua_error(L);
     } else {
         id = lua_tointeger(L, 1);
     }
+
     Host& host = getHostFromLua(L);
     TRoom* pR = host.mpMap->mpRoomDB->getRoom(id);
     if (!pR) {
-        lua_pushstring(L, "getRoomChar: room ID does not exist");
-        lua_error(L);
-        return 1;
+        lua_pushnil(L);
+        lua_pushfstring(L, "room with id %d does not exist", id);
+        return 2;
+    } else {
+        if (pR->mSymbol.isEmpty()) {
+            lua_pushstring(L, "");
+            return 1;
+        } else if (pR->mSymbol.length() >1) {
+            lua_pushstring(L, "?");
+            lua_pushfstring(L, "the symbol for the room cannot be represented as a single ASCII character, use getRoomSymbol(roomId) instead");
+            return 2;
+        } else if (pR->mSymbol.at(0).row()) {
+            lua_pushstring(L, "?");
+            lua_pushfstring(L, "the symbol for the room cannot be represented as a single ASCII character, use getRoomSymbol(roomId) instead");
+            return 2;
+        } else {
+            char character[2];
+            character[0] = pR->mSymbol.at(0).toLatin1();
+            character[1] = '\0';
+            lua_pushstring(L, character);
+            return 1;
+        }
+    }
+}
+
+int TLuaInterpreter::setRoomSymbol(lua_State* L)
```

Comment:
Well it is no longer a "char" a single ASCII/ISO 8859-1 byte/character - and that can break scripts using the getter which is also a no-no isn't it?  I've provided backwards compatibility - `getRoomChar` will return `?` if the symbol is not one that is expressible in that byte form...

Reply:
I think the chances of breaking scripts are pretty minimal, because we're going from one character to many and I don't think anyone would be triple-checking that the character(s) they get back are strictly ISO 8859-1 and erroring if it's anything else. Your thoughts, @keneanung?

We make use of `getRoomChar()` in the IRE mapping script and I think that will still work fine.

Reply:
This is still a point, I'm not a fan of telling people "yeah there's two ways you can set a symbol on a map good luck memorizing the right one"

Reply:
I thought the discussion was that "Symbol" was the most obvious/best term to use and `setRoomChar` is no longer a good fit for the term as - although neither implies a plural usage the way things are handled in the display usage does treat say the term where is it a word "shop" as a specific "symbol" and "cafe" and "café" as two different ones...

I would want to flag `setRoomChar` as __deprecated__ in the Wiki and for the current usage by the IRE scripts it would continue to do what it says it does (although it is now possible to clear the symbol which was not before!) - and we'd follow the previous behaviour in that if the `setRoomSymbol` function exists use it (you've always said that scripter should "test for a function rather than relying on a version number" 😮 )  otherwise fall-back to `setRoomChar` but it **only accepts a single ASCII character**...